frontend/lib/search/location_button.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/material.dart';
+import 'package:geolocator/geolocator.dart';
+
+import '../location/determine_position.dart';
+import '../widgets/small_button_spinner.dart';
+
+class LocationButton extends StatefulWidget {
+  final void Function(Position position) setCoordinates;
+
+  const LocationButton({Key key, this.setCoordinates}) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => _LocationButtonState();
+}
+
+enum LocationRequestStatus { requesting, requestSuccessful, requestFailed }
+
+class _LocationButtonState extends State<LocationButton> {
+  LocationRequestStatus _locationStatus = LocationRequestStatus.requesting;
+
+  @override
+  Widget build(BuildContext context) {
+    var theme = Theme.of(context);
+    if (_locationStatus == LocationRequestStatus.requestSuccessful) {
+      return Container();
+    }
+    var onPressed = _locationStatus == LocationRequestStatus.requesting
+        ? null
```

Comment:
Could that cause a null pointer exception?

Reply:
Nope, setting the onPressed handler to null is the way to disable the button on flutter 😅 I was surprised by that as well 🙈